### PR TITLE
Define fungible asset, token, and unit in the glossary

### PR DESCRIPTION
**Pull request by saharct**
### Description

Let's generically define these terms to consider current and future ways of creating fungible assets. We can rephrase or add a bullet if more specificity is needed, but we should first work toward clarifying any confusion within the standards docs.

### CODE REVIEW

File: developer-docs-site/docs/reference/glossary.md
Comment (lightmark) [inline developer-docs-site/docs/reference/glossary.md:193]: tokens usually means nft in our docs, which is not FA.

**Review by lightmark — COMMENTED** [review]
I think another term is `fungible token`?

File: developer-docs-site/docs/reference/glossary.md
Comment (saharct) [reply developer-docs-site/docs/reference/glossary.md:193]: But we also have docs where we refer to FTs as tokens. [AIP-21](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-21.md) uses terms like "tokenization" and "tokenized" to describe FAs. That being said, I'll remove the last bit from the second bullet point so that we can let the context in each doc decide.

**Comment by saharct** [comment]
> I think another term is `fungible token`?

Yes, I'll add `Fungible Token` too. I initially wanted to mention tokens as a potential type of FA to avoid a separate definition, but it's worth defining `Fungible Token` separately.

File: developer-docs-site/docs/reference/glossary.md
Comment (lightmark) [inline developer-docs-site/docs/reference/glossary.md:197]: this is different in token v2. fungible token should be a special type of fungible asset that the asset metadata is a token.

Comment (saharct) [reply developer-docs-site/docs/reference/glossary.md:197]: I defined v1 first since the v2 docs are not public yet. I'm okay with updating the definition in this PR to include v1 and v2, then waiting until the other docs land before landing this PR.

Comment (lightmark) [reply developer-docs-site/docs/reference/glossary.md:197]: Do we want to keep it in general?

File: developer-docs-site/docs/reference/glossary.md
Comment (davidiw) [inline developer-docs-site/docs/reference/glossary.md:194]:
```suggestion
- Follow the [Digital Asset Standards](../standards/index.md#digital-asset-standards) to create fungible assets on the Aptos blockchain. 
- Next generation of the Coin standard that addresses shortcomings of `aptos_framework::coin` such as lack of guaranteed enforcement of freeze and burn and advanced functionalities such as programmable transfers, e.g., approve in ERC-20.

```

File: developer-docs-site/docs/reference/glossary.md
Comment (davidiw) [reply developer-docs-site/docs/reference/glossary.md:197]: agreed, let's define two bullets here.

Comment (davidiw) [inline developer-docs-site/docs/reference/glossary.md:197]:
```suggestion
- For TokenV1 (aptos_token::token), a **fungible token** is a token that is interchangeable with other identical tokens (i.e., tokens that share the same `TokenId`). This means the tokens have the same `creator address`, `collection name`, `token name`, and `property version`.
- For TokenV2 (aptos_token_objects::token), a **fungible token** is a fungible asset with metadata object includes a TokenV2 resource.
```

**Review by davidiw — APPROVED** [review]

**Pull request merged**
